src/mono/wasm/debugger/BrowserDebugProxy/MemberObjectsExplorer.cs

```diff
@@ -150,8 +152,9 @@ private static string GetNamePrefixForValues(string memberName, string typeName,
                 {
                     // a collection - expose elements to be of array scheme
                     var memberNamedItems = members
-                        .Where(m => m["name"]?.Value<string>() == "Items" || m["name"]?.Value<string>() == "_items")
-                        .FirstOrDefault();
+                        .FirstOrDefault(m => m["name"]?.Value<string>() == "Items" || m["name"]?.Value<string>() == "_items");
+                    // sometimes items have dummy empty elements added after real items
+                    maxSize = members.FirstOrDefault(m => m["name"]?.Value<string>() == "_size")?["value"]?["value"]?.Value<int>();
```

Comment:
What is this doing? And who adds `_size`, and `_items`?

Reply:
Ok, so these are private fields for `List<T>`. We shouldn't depend on a type's private members like this. What do the "dummy empty elements" look like?

The error condition can be recreated by `a = new List<int>(capacity: 20); a.Add(1);` -> now the internal array `_items` will be of size 20, but it will have only 1 real value.

Reply:
Does VS show all the elements in `_items`?

Reply:
We have 2 behaviors in `DebuggerTests.EvaluateOnCallFrameTests.EvaluateBrowsableRootHidden`. The data:
```
[InlineData("EvaluateBrowsableClass", "TestEvaluateFieldsRootHidden", "testFieldsRootHidden", 10)]
[InlineData("EvaluateBrowsableClass", "TestEvaluatePropertiesRootHidden", "testPropertiesRootHidden", 10)]
[InlineData("EvaluateBrowsableStruct", "TestEvaluatePropertiesRootHidden", "testPropertiesRootHidden", 10)]
[InlineData("EvaluateBrowsableClassStatic", "TestEvaluateFieldsRootHidden", "testFieldsRootHidden", 10)]
[InlineData("EvaluateBrowsableClassStatic", "TestEvaluatePropertiesRootHidden", "testPropertiesRootHidden", 10)]
[InlineData("EvaluateBrowsableStructStatic", "TestEvaluateFieldsRootHidden", "testFieldsRootHidden", 10)]
[InlineData("EvaluateBrowsableStructStatic", "TestEvaluatePropertiesRootHidden", "testPropertiesRootHidden", 10)]
[InlineData("EvaluateBrowsableNonAutoPropertiesClass", "TestEvaluatePropertiesRootHidden", "testPropertiesRootHidden", 5)]
[InlineData("EvaluateBrowsableNonAutoPropertiesClassStatic", "TestEvaluatePropertiesRootHidden", "testPropertiesRootHidden", 5)]
```
result in producing the information you are talking about. On calling `GetObjectMemberValues` for `listRootHidden` in `GetRootHiddenChildren` we get a publicly available object that looks like this:
```
  "value": {
    "type": "object",
    "value": null,
    "description": "int[2]",
    "className": "int[]",
    "objectId": "dotnet:array:22",
    "subtype": "array"
  },
  "writable": false,
  "name": "Items",
  "isOwn": true,
  "__section__": "result",
  "__state__": null,
  "__parentTypeId__": 2,
  "__isNewSlot__": false
}
```
and it does not cause any problems: array has only two elements, it's public and always named `Items`, so after calling 
`JArray resultValue = await sdbHelper.GetArrayValues(rootObjectId.Value, token);` on it, we will get the items we should expose.
However for these cases:
```
[InlineData("EvaluateBrowsableStruct", "TestEvaluateFieldsRootHidden", "testFieldsRootHidden", 10)]
[InlineData("EvaluateBrowsableNonAutoPropertiesStruct", "TestEvaluatePropertiesRootHidden", "testPropertiesRootHidden", 5)]
[InlineData("EvaluateBrowsableNonAutoPropertiesStructStatic", "TestEvaluatePropertiesRootHidden", "testPropertiesRootHidden", 5)]
```
we are getting only following private objects:
```
{
  "value": {
    "type": "object",
    "value": null,
    "description": "int[4]",
    "className": "int[]",
    "objectId": "dotnet:array:46",
    "subtype": "array"
  },
  "writable": false,
  "isOwn": true,
  "name": "_items",
  "__state__": null,
  "__section__": "private"
}
{
  "value": {
    "type": "number",
    "value": 2,
    "description": "2"
  },
  "writable": true,
  "isOwn": true,
  "name": "_size",
  "__state__": null,
  "__section__": "private"
}
{
  "value": {
    "type": "number",
    "value": 2,
    "description": "2"
  },
  "writable": true,
  "isOwn": true,
  "name": "_version",
  "__state__": null,
  "__section__": "private"
}
{
  "value": {
    "type": "object",
    "value": null,
    "description": "int[0]",
    "className": "int[]",
    "objectId": "dotnet:array:66",
    "subtype": "array"
  },
  "writable": false,
  "isOwn": true,
  "name": "s_emptyArray",
  "__state__": null,
  "__section__": "private",
  "__isStatic__": true
}
```
The only way to get access to children of the hidden root is to call `GetArrayValues` on private `_items` object. As you can see, it has 4 elements even though the size of the list == 2, they are like this (empty elements always in the end):
```
{
  "value": {
    "type": "number",
    "value": 1,
    "description": "1"
  },
  "writable": true,
  "name": "0"
}
{
  "value": {
    "type": "number",
    "value": 2,
    "description": "2"
  },
  "writable": true,
  "name": "1"
}
{
  "value": {
    "type": "number",
    "value": 0,
    "description": "0"
  },
  "writable": true,
  "name": "2"
}
{
  "value": {
    "type": "number",
    "value": 0,
    "description": "0"
  },
  "writable": true,
  "name": "3"
}
```
In Console Application we see the same data when we expand private members.
![image](https://user-images.githubusercontent.com/32700855/195335552-876d759a-3611-47a9-9e78-4168dfc3ede6.png)

Reply:
> btw, you can use `list.Capacity` to know the size of `_items`.

In the screenshot above we can se it would not work, capacity is 4 even though our array has only 2 elements.

Reply:
> Ok, so these are private fields for `List<T>`. We shouldn't depend on a type's private members like this. What do the "dummy empty elements" look like?
> 
> The error condition can be recreated by `a = new List<int>(capacity: 20); a.Add(1);` -> now the internal array `_items` will be of size 20, but it will have only 1 real value.

I understand you suggest returning these additional "dummy elements" as well and changing the tests, not the proxy logic? If so, then I would disagree with it, in Console Application if an object contains a list that is decorated with `RootHidden` we see only two elements on evaluation:
![image](https://user-images.githubusercontent.com/32700855/195337036-de576d7c-f301-4ea3-90e0-7ec9c1825141.png)